api/src/services/users/users.ts

```diff
@@ -10,13 +11,53 @@ import { AuthenticationError } from '@redwoodjs/graphql-server'
 import { ROLES } from 'src/lib/constants'
 import { db } from 'src/lib/db'
 
+export const currentUserIsUSDRAdmin = (): boolean => {
+  return context.currentUser?.roles?.includes(ROLES.USDR_ADMIN)
+}
+
 export const users: QueryResolvers['users'] = () => {
-  return db.user.findMany()
+  if (currentUserIsUSDRAdmin()) {
+    return db.user.findMany()
+  }
+
+  const userAgency = context.currentUser.agency as Agency
+  validate(userAgency.organizationId, {
```

Comment:
We can remove all these validations when we make organization non-nullable on agency, see comment here: https://usdigitalresponse.slack.com/archives/C056HNBK12T/p1710271751319379?thread_ts=1710271575.786199&cid=C056HNBK12T